File: iOSApp/iOSApp/Location.swift
//
//  Location.swift
//  iOSApp
//
//  Created by Peter Hitchcock on 3/2/16.
//  Copyright © 2016 Peter Hitchcock. All rights reserved.
//

import Foundation

class Location {
    var name: String!
    var streetNumber: String!
    var streetName: String!
    var city: String!
    var state: String!
    var zip: String!
    var date: String?
    var time: String?
    var createdAt: String?
    var updatedAt: String?

    init(name: String, streetNumber: String, streetName: String, city: String, state: String, zip: String) {

        self.name = name
        self.streetNumber = streetNumber
        self.streetName = streetName
        self.city = city
        self.state = state
        self.zip = zip
    }

}

File: RailsApp/app/models/location.rb
# == Schema Information
#
# Table name: locations
#
#  id            :integer          not null, primary key
#  name          :string
#  street_number :string
#  street_name   :string
#  city          :string
#  state         :string
#  zip           :string
#  date          :string
#  time          :string
#  created_at    :datetime         not null
#  updated_at    :datetime         not null
#  info          :text
#

class Location < ActiveRecord::Base
   validates :name, :street_number, :street_name, :city, :state, :zip, :date, :time, :info, presence: true

   geocoded_by :address
   after_validation :geocode

   def address
    ["#{street_number} #{street_name}", city, state, zip].compact.join(', ')
   end
end
